src/content/en/fundamentals/engage-and-retain/push-notifications/how-push-works.md

```diff
@@ -0,0 +1,160 @@
+project_path: /web/_project.yaml
+book_path: /web/fundamentals/_book.yaml
+
+{# wf_updated_on: 2017-03-03 #}
+{# wf_published_on: 2016-06-30 #}
+
+# How Push Works {: .page-title }
+
+{% include "web/_shared/contributors/mattgaunt.html" %}
+
+
+
+Before getting into the API, let's look at push from a highlevel, start to finish. Then as we
+step through individual topics or API's later on, you'll have an idea of how and why it's
+important.
+
+The three key steps when implementing push are:
+
+1. Adding the client side logic to subscribe a user to push (i.e. the JavaScript and UI in your
+web app that registers a user to push messages).
+1. The API call from your back-end / application that triggers a push message  to a users device.
+1. The service worker JavaScript file that will receive a "push event" when the push arrives on
+the device. It's in this JavaScript that you'll be able to show a notification.
+
+Let's look at what each of these steps entails in a little more detail.
+
+## Step 1: Client Side
```

Comment:
Please make headings 2 and below sentence case (but without a period.) Please verify throughout.